File: eCar/eCar.Services/Database/Request.cs
using System;
using System.Collections.Generic;

namespace eCar.Services.Database;

public partial class Request
{
    public int Id { get; set; }

    public bool IsAccepted { get; set; }

    public int RouteId { get; set; }

    public int DriverId { get; set; }

    public virtual Driver Driver { get; set; } = null!;

    public virtual Route Route { get; set; } = null!;
}

File: eCar/UI/ecar_mobile/lib/models/Request/request.dart
import 'package:ecar_mobile/models/Driver/driver.dart';
import 'package:ecar_mobile/models/Route/route.dart';
import 'package:json_annotation/json_annotation.dart';

part 'request.g.dart';

@JsonSerializable()
class Request {
  int? id;
  bool? isAccepted;
  int? routeId;
  int? driverId;
  Driver? driver;
  Route? route;
  Request(
      {this.id,
      this.isAccepted,
      this.routeId,
      this.driverId,
      this.driver,
      this.route});

  factory Request.fromJson(Map<String, dynamic> json) =>
      _$RequestFromJson(json);

  Map<String, dynamic> toJson() => _$RequestToJson(this);
}
